### PR TITLE
Added create-mirror-UI

### PR DESCRIPTION
**Pull request by mayankkapoor**
Instead of API, a user can now also create a mirror with a UI
- Added MirrorDefinitionForm to models
- Added new Mirror definition template
- Added New button to mirrors.html
- Added url for new mirror to urls.py
- Added mirror_definition function to views.py

### CODE REVIEW

**Comment by mayankkapoor** [comment]
@sorenh I think the circleCI tests failed due to an issue with circleCI, not with the code. Please check if this can be merged.

**Comment by sorenisanerd** [comment]
Yup, I've fixed the CircleCI problem. It has to do with my moving some of the requirements into test-requirements.txt.

File: aasemble/django/apps/mirrorsvc/models.py [hunk 1]
```diff
@@ -160,3 +161,9 @@ def save(self, *args, **kwargs):
     def perform_snapshot(self):
         self.sync_dists()
         self.symlink_pool()
+
+
+class MirrorDefinitionForm(ModelForm):
+    class Meta:
+        model = Mirror
+        fields = ['url', 'series', 'components', 'public']
```
Comment (sorenisanerd) [inline aasemble/django/apps/mirrorsvc/models.py]: I don't think the form belongs in models.py. Move it to a new forms.py or views.py (your choice).

I know there's another Form in buildsvc's models.py, but that's also a mistake. I'll move it somewhere else.

**Comment by mayankkapoor** [comment]
@sorenh Ready for review

**Comment by sorenisanerd** [comment]
This is live now.

**Pull request merged**